src/stitching/delegateToSchema.ts

```diff
@@ -22,113 +22,148 @@ import FilterToSchema from '../transforms/FilterToSchema';
 import AddTypenameToAbstract from '../transforms/AddTypenameToAbstract';
 import CheckResultAndHandleErrors from '../transforms/CheckResultAndHandleErrors';
 
-export default async function delegateToSchema(
-  options: IDelegateToSchemaOptions,
-): Promise<any> {
-  const { info, args = {} } = options;
-  const rawDocument: DocumentNode = createDocument(
-    options.fieldName,
-    options.operation,
-    info.fieldNodes,
-    Object.keys(info.fragments).map(
-      fragmentName => info.fragments[fragmentName],
-    ),
-    info.operation.variableDefinitions,
+export function createBatchOperation(
+  targetSchema: GraphQLSchema,
+  targetOperation: 'query' | 'mutation' | 'subscription',
+  rootDefs: { [key: string]: [{ [key: string]: any }, { [key: string]: any }] },
+  graphqlContext: { [key: string]: any },
+  documentInfo: {
+    operation: {
+      name?: { [key: string]: any }
+      variableDefinitions?: Array<VariableDefinitionNode>,
+    },
+    variableValues?: { [variableName: string]: any },
+    fragments?: { [fragmentName: string]: FragmentDefinitionNode },
+  },
+  transforms?: Array<Transform>,
+): FetcherOperation {
+  const roots = Object.keys(rootDefs).map(key => {
```

Comment:
Question here is whether we should auto-alias stuff or if we should expect user to alias themselves. I guess latter. Data-loading API can then alias by itself.

Reply:
I was thinking users should add aliasing themselves:
```js
{
  'user1:node': [{ id: 1 }, ...],
  'user2:node': [{ id: 2 }, ...]
}
```